File: app/src/main/java/il/ronmad/speedruntimer/CategoryAdapter.java
package il.ronmad.speedruntimer;

import android.content.Context;
import android.support.v4.content.ContextCompat;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

public class CategoryAdapter extends MyBaseAdapter<Category> {

    public CategoryAdapter(Context context, List<Category> categories) {
        super(context, categories);
        this.listItemResourceId = R.layout.category_list_item;
    }

    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {
        View listItemView = super.getView(i, view, viewGroup);

        TextView nameText = listItemView.findViewById(R.id.categoryName);
        TextView bestTimeText = listItemView.findViewById(R.id.pbTime);
        TextView runCountText = listItemView.findViewById(R.id.runsNum);

        Category category = getItem(i);
        nameText.setText(category.name);
        bestTimeText.setText(category.bestTime > 0 ? Util.getFormattedTime(category.bestTime) : "None yet");
        bestTimeText.setTextColor(ContextCompat.getColor(context,
                category.bestTime > 0 ? R.color.colorAccent : android.R.color.primary_text_light));
        runCountText.setText(String.valueOf(category.runCount));

        return listItemView;
    }
}

File: app/src/main/kotlin/il/ronmad/speedruntimer/CategoryAdapter.kt
package il.ronmad.speedruntimer

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import kotlinx.android.synthetic.main.category_list_item.view.*

class CategoryAdapter(val context: Context, categories: List<Category>)
    : BaseRecyclerViewAdapter<Category>(categories) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
            CategoryViewHolder(LayoutInflater.from(parent.context)
                    .inflate(R.layout.category_list_item, parent, false))

    override fun onBindViewHolder(holder: BaseViewHolder<Category>, position: Int) {
        super.onBindViewHolder(holder, position)
        (holder as CategoryViewHolder).apply {
            val category = item
            categoryNameText.text = category.name
            pbTime.text = if (category.bestTime > 0) category.bestTime.getFormattedTime() else "None yet"
            pbTime.setTextColor(context.getColorCpt(if (category.bestTime > 0) R.color.colorAccent
            else android.R.color.primary_text_light))
            runsNum.text = category.runCount.toString()
        }
    }

    override fun onItemMoved(oldPos: Int, newPos: Int) {
        notifyItemMoved(oldPos, newPos)
    }

    override fun onItemEdited(position: Int) {
        notifyItemChanged(position)
    }

    class CategoryViewHolder(categoryView: View) : BaseViewHolder<Category>(categoryView) {
        val categoryNameText: TextView = categoryView.categoryName
        val pbTime: TextView = categoryView.pbTime
        val runsNum: TextView = categoryView.runsNum
    }
}
